### PR TITLE
ci: add workflow linter

### LINKED ISSUE #65: add yaml linter for workflow files
The `.yml` workflow files do not have consistent formatting (esp. w/r/t newlines).  We should add a linter and an action to enforce consistent formatting.

### PR DESCRIPTION
**Pull request by wexxlee**
Closes #65 

This adds a linter and corresponding workflow to lint GitHub workflow files.  It has three components:

1. [yamllint](https://pypi.org/project/yamllint/), which handles general YAML syntax lint and some formatting
2. A custom linting script, `.github/scripts/lint_workflow.cjs`, which enforces some specific constraints around workflow file formatting, especially newlines and block ordering.  (I looked for existing tools to provide this functionality but came up empty; newline usage is not governed by YAML schema, so enforcing it is something we'll have to do on our own.)
3. [actionlint](https://github.com/rhysd/actionlint), which provides extensive checking specific to GitHub workflow scripts, including things like semantic errors, shellcheck, dependency checking, etc.

This also adds a workflowlint script command (e.g. `npm run workflowlint`).

### CODE REVIEW

**Comment by wexxlee** [comment]
A few last-minute tweaks, but this should be ready for code review if anyone has a chance to look at it.

File: .github/workflows/workflow-lint.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+          node .github/scripts/lint-workflow.cjs
+        env:
+          GH_TOKEN: ${{ secrets.BOT_TOKEN }}
+
+      - name: Download actionlint
+        id: get_actionlint
+        run: |
+          bash <(curl \
+          https://raw.githubusercontent.com/rhysd/actionlint/main/scripts/download-actionlint.bash)
+        shell: bash
```
Comment (valarnin) [inline .github/workflows/workflow-lint.yml:42]: I'm a bit leery of a raw download and execute of a rolling file target like this. Can we target the bash script to a specific commit (preferably the commit corresponding to the tag for the release we're using) and target a specific version?

File: .github/scripts/lint-workflow.cjs [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+      ghErrOutput += `${err}\n`;
+    });
+    ghErrOutput += '::endgroup::\n';
+
+    console.log(`Found ${fileErrors.length} errors in ${currFile}.`);
+  }
+  fileErrors = [];
+};
+
+const parseFile = (file) => {
```
Comment (valarnin) [inline .github/scripts/lint-workflow.cjs:58]: This is a really big function but I don't really see a sane way to break it up into subfunctions?

**Review by valarnin — APPROVED** [review]

File: .github/workflows/workflow-lint.yml (hunk 1 above)
Comment (wexxlee) [reply .github/workflows/workflow-lint.yml:42]: > I'm a bit leery of a raw download and execute of a rolling file target like this. Can we target the bash script to a specific commit (preferably the commit corresponding to the tag for the release we're using) and target a specific version?

I think there's an even better option - it looks like the project has docker support, so we can run it out of docker without ever needing to grab the executable.

File: .github/scripts/lint-workflow.cjs (hunk 2 above)
Comment (wexxlee) [reply .github/scripts/lint-workflow.cjs:58]: > This is a really big function but I don't really see a sane way to break it up into subfunctions?

Yeah, unfortunately there's not a lot of repeated code, just a lot of logic steps.  At least it's commented? 😁

**Comment by valarnin** [comment]
This looks good to me now. Feel free to merge at your convenience.

**Pull request merged**